src/vuex/modules/index/store.ts

```diff
@@ -1,230 +1,172 @@
+  removeCollection(state, { index, collection }: IndexCollectionPayload): void {
+    state.indexes[getIndexPosition(state.indexes, index.name)].removeCollection(
+      collection
+    )
   }
 })
 
 const actions = createActions({
-  async createIndex(context, index: string) {
+  async createIndex(context, name: string) {
     const { commit, rootGetters } = indexActionContext(context)
 
-    await rootGetters.kuzzle.$kuzzle.index.create(index)
+    await rootGetters.kuzzle.$kuzzle.index.create(name)
+    const index = new Index(name)
+
     commit.addIndex(index)
   },
   async deleteIndex(context, index) {
     const { commit, rootGetters } = indexActionContext(context)
 
-    await rootGetters.kuzzle.$kuzzle.index.delete(index)
-    removeIndex(index)
-    commit.deleteIndex(index)
+    await rootGetters.kuzzle.$kuzzle.index.delete(index.name)
+
+    commit.removeIndex(index)
   },
   async listIndexes(context) {
     const { commit, rootGetters } = indexActionContext(context)
+    const indexes = state.indexes
     commit.setLoadingIndexes(true)
+
     let result = await rootGetters.kuzzle.$kuzzle.index.list()
-    result = result.filter(index => index !== '%kuzzle')
+    result = result.filter(
+      (indexName: string) => indexName !== INTERNAL_INDEX_NAME
+    )
 
-    for (const index of result) {
-      commit.addIndex(index)
+    for (const indexName of result) {
+      // we only need to add the indexes not yet fetched
+      if (getIndexPosition(state.indexes, indexName) === -1) {
+        commit.addIndex(new Index(indexName))
+      }
     }
 
     commit.setLoadingIndexes(false)
   },
-  async listCollectionsForIndex(context, index) {
+  async fetchCollections(context, index: Index) {
     const { commit, rootGetters } = indexActionContext(context)
-    commit.setLoadingCollectionsForIndex({ index, loading: true })
-    const res = await rootGetters.kuzzle.$kuzzle.collection.list(index, {
-      size: 0
-    })
-    // debugger
-    let collections = splitRealtimeStoredCollections(res.collections)
-
-    if (!collections.realtime) {
-      collections.realtime = []
-    }
+    commit.setLoadingCollections({ index, loading: true })
 
-    collections.realtime = collections.realtime.concat(
-      getRealtimeCollectionFromStorage(index)
-    )
-    collections = dedupeRealtimeCollections(collections)
-    commit.setCollectionsForIndex({
-      index,
-      collections: collections.stored,
-      type: 'stored'
-    })
-    commit.setCollectionsForIndex({
-      index,
-      collections: collections.realtime,
-      type: 'realtime'
-    })
+    const result = await rootGetters.kuzzle.$kuzzle.collection.list(index.name)
 
-    commit.setLoadingCollectionsForIndex({ index, loading: false })
-  },
+    const collections = result.collections.array.map(el => {
+      const collectionType =
+        el.type === CollectionType.STORED
+          ? CollectionType.STORED
+          : CollectionType.REALTIME
```

Comment:
This implies that a collection of an unknown type (like, in the future) will be treated as a `REALTIME` one. You should use this check to trigger an exception if the type is unknown.